crates/consensus/src/transaction/rlp.rs

```diff
@@ -195,8 +195,8 @@ pub trait RlpEcdsaDecodableTx: RlpEcdsaEncodableTx {
 
     /// Decodes the transaction from eip2718 bytes, expecting the default type
     /// flag.
-    fn eip2718_decode(buf: &mut &[u8]) -> Eip2718Result<Signed<Self>> {
-        Self::eip2718_decode_with_type(buf, Self::DEFAULT_TX_TYPE)
+    fn eip2718_decode(&self, buf: &mut &[u8]) -> Eip2718Result<Signed<Self>> {
```

Comment:
we can't give the decoding fns a &self arg because this defeats the purpose of decoding, so we need to fix this separately